src/js/components/ServiceForm.js

```diff
@@ -233,8 +233,37 @@ class ServiceForm extends SchemaForm {
 
   updateDefinitions() {
     let {model} = this.internalStorage_get();
-    let {networking} = this.multipleDefinition;
+    let {networking, volumes} = this.multipleDefinition;
+    let {containerSettings} = model;
 
+    let showDockerVolumes = containerSettings &&
+      containerSettings.image != null && containerSettings.image !== '';
+
+    let endIndex = -1;
```

Comment:
I really don't like that we rely on these indices, but I can't think of a better solution without refactoring huge parts of form. Let's make sure we get rid of these once we refactor the schema form.